peekingduck/cli.py

```diff
@@ -86,19 +86,29 @@ def init(custom_folder_name: str) -> None:
         "List of nodes to run. None assumes run_config.yml at current working directory"
     ),
 )
+@click.option(
+    "--log_level",
+    default="info",
+    help="""Modify log level {"critical", "error", "warning", "info", "debug"}""",
+)
 @click.option(
     "--node_config",
     default="None",
     help="""Modify node configs by wrapping desired configs in a JSON string.\n
         Example: --node_config '{"node_name": {"param_1": var_1}}'""",
 )
 @click.option(
-    "--log_level",
-    default="info",
-    help="""Modify log level {"critical", "error", "warning", "info", "debug"}""",
+    "--num_iter",
+    default=None,
+    type=int,
+    help="""Stop pipeline after running this number of iterations""",
 )
 def run(
-    config_path: str, node_config: str, log_level: str, nodes_parent_dir: str = "src"
+    config_path: str,
+    log_level: str,
+    node_config: str,
+    num_iter: int,
```

Comment:
Similarly here, `config_path` and `node_config` can both also be `None`.
I think while the `Optional[x]` is pedantic, it makes the function signature overcrowded with optional stuff (sorry, can't resist this pun). I'm also not sure about the realworld value of `Optional` in helping with linter checks and runtime type checking.
This is related to all the below remarks regarding `Optional[int]`.

Reply:
I think initially I tried to convert the other arguments to use `Optional` but gave up when the required amount of changes was spread across too many places. We can keep it as it is since the linter doesn't complain.

Personally I find type hinting as `Optional`, i.e., this variable can be `None`, to be helpful in avoiding some `TypeError` during development. For example, `None == 0` works but `None >= 0` raises `TypeError`.